editor/src/clj/editor/code/view.clj

```diff
@@ -1025,6 +1027,240 @@
+            lsp (lsp/get-node-lsp (:basis evaluation-context) resource-node)]
+        (swap-state assoc :done true)
+        (lsp/rename
+          lsp
+          rename-cursor-range
+          text
+          (fn on-rename-response [resource->ascending-cursor-ranges-and-replacements]
+            (ui/run-later
+              (some->
+                (g/with-auto-evaluation-context evaluation-context
+                  (when (identical? rename-cursor-range (get-property view-node :rename-cursor-range evaluation-context))
+                    (let [resource->view (->> (get-property view-node :open-views evaluation-context)
+                                              (e/keep
+                                                (fn [[view {:keys [resource]}]]
+                                                  (when (g/node-kw-instance? (:basis evaluation-context) ::CodeEditorView view)
+                                                    [resource view])))
+                                              (into {}))
+                          project (get-property view-node :project evaluation-context)]
+                      (into (set-properties view-node nil {:rename-cursor-range nil})
+                            (mapcat
+                              (fn [[resource ascending-cursor-ranges-and-replacements]]
+                                (when-let [resource-node (project/get-resource-node project resource evaluation-context)]
+                                  (when (g/node-instance? (:basis evaluation-context) r/CodeEditorResourceNode resource-node)
+                                    (if-let [view (resource->view resource)]
+                                      (set-properties
+                                        view nil
+                                        (data/apply-edits
+                                          (get-property view :lines evaluation-context)
+                                          (get-property view :regions evaluation-context)
+                                          (get-property view :cursor-ranges evaluation-context)
+                                          ascending-cursor-ranges-and-replacements
+                                          (get-property view :layout evaluation-context)))
+                                      (set-resource-properties
+                                        resource-node
+                                        (data/apply-edits
+                                          (g/node-value resource-node :lines evaluation-context)
+                                          (g/node-value resource-node :regions evaluation-context)
+                                          (g/node-value resource-node :cursor-ranges evaluation-context)
+                                          ascending-cursor-ranges-and-replacements)))))))
+                            resource->ascending-cursor-ranges-and-replacements))))
```

Comment:
Perform a rename, but we have separate code paths for 2 different cases:
1. when we have the view, we apply the edits and adjust the view layout
2. when we don't have a view, we apply resource edits.